core/src/test/java/io/confluent/rest/SaslTest.java

```diff
@@ -0,0 +1,262 @@
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.TrustSelfSignedStrategy;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.apache.http.ssl.SSLContexts;
+import org.apache.http.util.EntityUtils;
+import org.apache.kafka.common.config.types.Password;
+import org.apache.kafka.test.TestSslUtils;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import sun.security.provider.certpath.OCSPResponse;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.net.SocketException;
+import java.nio.file.Files;
+import java.nio.file.OpenOption;
+import java.nio.file.StandardOpenOption;
+import java.security.KeyPair;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.SSLException;
+import javax.net.ssl.SSLHandshakeException;
+import javax.security.auth.login.Configuration;
+import javax.ws.rs.DefaultValue;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Configurable;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+
+import io.confluent.common.metrics.KafkaMetric;
+import io.confluent.rest.annotations.PerformanceMetric;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+
+public class SaslTest {
+  private static final Logger log = LoggerFactory.getLogger(SaslTest.class);
+
+  private File jaasFile;
+  private File loginPropertiesFile;
+  private String previousAuthConfig;
+  private SaslTestApplication app;
+  private CloseableHttpClient httpclient;
+  String httpUri = "http://localhost:8080";
+
+  @Before
+  public void setUp() throws Exception {
+    jaasFile = File.createTempFile("jaas", ".config");
+    loginPropertiesFile = File.createTempFile("login", ".properties");
+
+    String jaas = "c3 {\n"
+                  + "  org.eclipse.jetty.jaas.spi.PropertyFileLoginModule required\n"
+                  + "  debug=\"true\"\n"
+                  + "  file=\"" + loginPropertiesFile.getAbsolutePath() + "\";\n"
+                  + "};\n";
+    Files.write(
+        jaasFile.toPath(),
+        jaas.getBytes(),
+        StandardOpenOption.TRUNCATE_EXISTING
+    );
+
+    String loginProperties = "jay: kafka,Administrators\n"
+                             + "neha: akfak,Administrators\n"
+                             + "jun: kafka-\n";
```

Comment:
Not sure how @nehanarkhede is going to feel about being an Administrator of `akfak`...